// @smic-applabs/db-gift/gift/schemas/skuSchema.js
const getSchemas = Schema => {   
      const BrandSchema = new Schema({
        name: String, 
        magentoId: String
      })

     const SkuSchema = new Schema({
       magentoId: String,
       skuCode: { type: String }, // { type: String, index: true }, Extracted for external index
       brand: String,
       skuBarcode: { type: String }, // { type: String, index: true }, Extracted for external index
       skuBarcodes: { type: [String] },
       babySkuCode: { type: String }, // { type: String, index: true }, Extracted for external index
       motherSkuCode: { type: String }, // { type: String, index: true }, Extracted for external index
       productName: { type: String }, // { type: String, index: true }, Extracted for external index
       sizeCode: String,
       sizeName: String,
       currentPrice: { min: 0.001, type: Number, required: true }, // { min: 0.01, type: Number, index: true, required: true }, Extracted for external index //currentPrice: { min: 0.001, type: Number, index: true, required: true } from both estore-graph and sm-store-job
       referencePrice: Number,
       salePrice: Number,
       departmentCode: String,
       departmentName: String,
       subDepartmentCode: String,
       subDepartmentName: String,
       classCode: String,
       className: String,
       environmentCode: String,
       categories: [String],
       discount: { type: Number, default: 0 }, // { type: Number, default: 0, index: true }, Extracted for external index
       status: { type: String, default: "Pending" }, // Pending, Active, Inactive, Discontinued
       description: String,
       rating: String,
       shortDescription: String,
       commercialName: { type: String }, // { type: String, index: true }, //productname (front-end) Extracted for external index
       brandName: { type: String }, // { type: String, index: true }, Extracted for external index
       tags: [String],
       thumbnails: [{
        url: String,
        size: Number,
        isOptimized: { type: Boolean, default: false }
       }],
       images: [{
        url: String,
        size: Number,
        isOptimized: { type: Boolean, default: false }
      }],
      isGiftBasket: { type: Boolean, default: true },
      lastUpdate: { type: Date, default: Date.now },
      createdDate: { type: Date, default: Date.now },
      maxQty: Number,
      syncId: String
     })
     
     return { BrandSchema, SkuSchema }
   }
   
   module.exports = getSchemas
