// @smic-applabs/db-gift/gift/schemas/giftBasketSchema.js
const getSchemas = (Schema) => {

  const GiftBasketSchema = new Schema({
    isCustom: { type: Boolean, default: false },
    name: { type: String, required: true },
    owner: { type: Schema.Types.ObjectId },
    deviceId: { type: String, required: true },
    createdDate: { type: Date, required: true , default: Date.now },
    updateDate: { type: Date, required: true , default: Date.now },
    status: { type: String, required: true , default: "Pending" },
  })

  const GiftBasketItemSchema = new Schema({
    giftBasket: { type: Schema.Types.ObjectId, required: true },
    sku: { type: Schema.Types.ObjectId, required: true },
    owner: { type: Schema.Types.ObjectId },
    deviceId: { type: String, required: true },
    quantity: { type: Number, default: 1 },
    addedDate: { type: Date, default: Date.now },
    status: { type: String, required: true , default: "Pending" },
  })

  const GiftBasketGifteeSchema = new Schema({
    owner: { type: Schema.Types.ObjectId },
    deviceId: { type: String, required: true },
    giftBasket: { type: Schema.Types.ObjectId, required: true },
    giftee: { type: Schema.Types.ObjectId, required: true },
    addedDate: { type: Date, default: Date.now },
    branch: { type: Schema.Types.ObjectId },
    quantity: { type: Number, default: 1 },
    deliveryDate: { type: Date },
  })

  return { GiftBasketSchema, GiftBasketItemSchema, GiftBasketGifteeSchema }
}

module.exports = getSchemas
